crates/solver/src/settlement_rater.rs

```diff
@@ -244,18 +248,251 @@ impl SettlementRating for SettlementRater {
             None => Score::Protocol(big_rational_to_u256(&objective_value).unwrap_or_default()),
         };
 
+        // recalculate score if success probability is provided
+        let score = match settlement.success_probability {
```

Comment:
Can we combine with the above match statement to make priorities between probability and scores very clear? Something like

match (settlement.success_probability, &settlement.score) {
  (Some(p), _) => ...;
  (None, Some(score) => ..;
  (None, None) => ...;
}

Reply:
It get's even messier if merged into single match.
Also, with the way it's done right now, droping `score` as part of https://github.com/cowprotocol/services/issues/1821 will be way easier (erasing one match).